module/entities/helper.js

```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
+        try {
+
+            let telemetry = [];
+            
+            let entityDocuments = await this.entityDocuments({
+                    _id: {
+                      $in : entityId
+                    }
+                }, [
+                        "_id",
+                        "metaInformation",
+                        "entityType",
+                        "entityTypeId",
+                        "updatedAt",
+                        "createdAt",
+                    ]);
+            
+            await Promise.all(entityDocuments.map(async entityDocument => {
+
+            let entityObj = {
+                _id: entityDocument._id,
+                entityType: entityDocument.entityType,
+                entityTypeId: entityDocument.entityTypeId,
+                updatedAt: entityDocument.updatedAt,
+                createdAt: entityDocument.createdAt
+            }
+            
+            for (let metaData in entityDocument.metaInformation) {
+                entityObj[metaData] = entityDocument.metaInformation[metaData];
+            }
+
+            let telemetryObj = {
+                [`${entityObj.entityType}_name`]: entityObj.name,
+                [`${entityObj.entityType}_id`]: entityObj._id,
+                [`${entityObj.entityType}_externalId`]: entityObj.externalId
+            };
+
+            let relatedEntities = await this.relatedEntities(
+                     entityObj._id,
+                     entityObj.entityTypeId,
+                     entityObj.entityType,
+                     [
+                        "metaInformation.externalId",
+                        "metaInformation.name",
+                        "entityType",
+                        "entityTypeId",
+                        "_id" 
+                     ])
+ 
+                  if (relatedEntities.length > 0) {
+
+                    relatedEntities = relatedEntities.map(entity => {
+          
+                      telemetryObj[`${entity.entityType}_name`] =
+                        entity.metaInformation.name;
+          
+                      telemetryObj[`${entity.entityType}_id`] =
+                        entity._id;
+          
+                      telemetryObj[`${entity.entityType}_externalId`] =
+                        entity.metaInformation.externalId;
+          
+                      return {
+                        name: entity.metaInformation.name,
+                        externalId: entity.metaInformation.externalId,
+                        entityType: entity.entityType,
+                        entityTypeId: entity.entityTypeId,
+                        _id: entity._id
+                      }
+                    })
+          
+                    entityObj["relatedEntities"] = relatedEntities;
+                  }
+
+                  telemetry.push(telemetryObj);
+
+                  entityObj["telemetry_entities"] = telemetry;
+
+            await elasticSearchData.createOrUpdate(
```

Comment:
@deepakr123 - Do format the code with indetation

Reply:
resolved